compiler/circle-input-names/src/circle-input-names.test.cpp

```diff
@@ -0,0 +1,124 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <vector>
+#include <regex>
+
+class circle_input_names_test : public ::testing::Test
+{
+protected:
+  void SetUp(void) override;
+  std::vector<std::string> get_input_names_of(std::string op);
+
+protected:
+  std::string _input_names_dict_str;
+};
+
+void circle_input_names_test::SetUp(void)
+{
+  std::string cmd = std::getenv("CIRCLE_INPUT_NAMES_PATH");
+  if (cmd.empty())
+  {
+    std::runtime_error("CIRCLE_INPUT_NAMES_PATH is not found");
+    return;
+  }
+
+  FILE *fp = popen(cmd.c_str(), "r");
+  if (!fp)
+  {
+    std::runtime_error("popen() failed");
+    return;
+  }
+
+  char buff[1024];
+  std::string result = "";
+  try
+  {
+    while (fgets(buff, sizeof(buff), fp) != NULL)
+    {
+      result += buff;
+    }
+  }
+  catch (...)
+  {
+    pclose(fp);
+    throw;
+  }
+
+  _input_names_dict_str = result;
+  pclose(fp);
+
+  return;
+}
+
+std::vector<std::string> circle_input_names_test::get_input_names_of(std::string op)
+{
+  std::vector<std::string> input_names{};
+
+  // Find op string key from _input_names_dict_str and parse the values input input_names vector
+  size_t pos = _input_names_dict_str.find(op);
+  if (pos == std::string::npos)
+  {
+    return input_names;
+  }
+  else
+  {
+    std::string substr = _input_names_dict_str.substr(pos);
+    size_t start_pos = substr.find("[");
+    size_t end_pos = substr.find("]");
+    if (start_pos != std::string::npos && end_pos != std::string::npos)
+    {
+      std::string names = substr.substr(start_pos + 1, end_pos - start_pos - 1);
+      std::stringstream ss(names);
+      std::string name;
+      while (std::getline(ss, name, ','))
+      {
+        std::smatch match;
+        std::regex pattern = std::regex(R"(^\s*\"([^\"]+)\"\s*$)");
+        if (std::regex_match(name, match, pattern))
+        {
+          input_names.push_back(match[1].str());
+        }
+      }
+    }
+  }
+  return std::move(input_names);
+}
+
+TEST_F(circle_input_names_test, valid_command) { ASSERT_FALSE(_input_names_dict_str.empty()); }
+
+TEST_F(circle_input_names_test, valid_names_softmax)
+{
+  // "SOFTMAX" should have single input: "logits"
+  auto names = get_input_names_of("SOFTMAX");
+  ASSERT_EQ(names.size(), 1);
+  ASSERT_EQ(names[0], "logits");
+}
+
+TEST_F(circle_input_names_test, valid_names_conv2d)
+{
+  // "CONV_2D" should have three inputs: "input", "filter", "bias"
+  auto names = get_input_names_of("CONV_2D");
+  ASSERT_EQ(names.size(), 3);
+  ASSERT_EQ(names[0], "input");
+  ASSERT_EQ(names[1], "filter");
+  ASSERT_EQ(names[2], "bias");
+}
```

Comment:
for our SE requirements, we need to add num(positive test) <= num(negative test).
please add some negative tests with name having `_NEG` suffix.

Reply:
strictly speaking, this(circle-input-names) is not included in our release module,
so negative tests are not counted. :/
but for the quality, negative tests are welcome :)

Reply:
Okay, I'll update it.

Reply:
Applied!